File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
import math

def solution(progresses, speeds):
    answer = []
    
    queue = deque()
    len_progress = len(progresses)
    for i in range(len_progress) :
        left_day = math.ceil((100-progresses[i]) / speeds[i])
        queue.append(left_day)
    
    while queue :
        tmp = 1
        current_day = queue.popleft()
        
        while queue and queue[0] <= current_day :
            queue.popleft()
            tmp += 1
            
        answer.append(tmp)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        int n = progresses.length;
        
        Queue<Integer> queue = new LinkedList<>();
        for (int i=0; i<n; i++) {
            int tmp = (100-progresses[i]) / speeds[i];
            if ((100-progresses[i]) % speeds[i] == 0) {
                queue.offer(tmp);
            } else {
                queue.offer(tmp+1);
            }
        }
        // System.out.println(queue);
        
        
        while (!queue.isEmpty()) {
            int value = queue.poll();
            // System.out.println(value);
            
            int cnt = 1;
            while (!queue.isEmpty() && queue.peek() <= value) {
                queue.poll();
                cnt++;
            }
            answer.add(cnt);
        }
        
        int[] result = new int[answer.size()];
        for (int i=0; i<answer.size(); i++) {
            result[i] = answer.get(i);
        }
        
        return result;
    }
}